src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -70,10 +70,30 @@ case object ThurloeDatabaseConnector extends DataAccess {
     } yield result
   }
 
+  def lookup(userId: String, key: String): Future[UserKeyValuePair] = lookupIncludingDatabaseId(userId, key) map { _.userKeyValuePair }
+
   def lookup(userId: String): Future[UserKeyValuePairs] = {
     for {
       results <- lookupWithConstraint(x => x.userId === userId)
-    } yield UserKeyValuePairs(userId, results map { _.keyValuePair })
+    } yield UserKeyValuePairs(userId, results map { _.userKeyValuePair.keyValuePair })
+  }
+
+  def lookup(queryParameters: ThurloeQuery): Future[Seq[UserKeyValuePair]] = {
+    for {
+      filteredOnUserAndKey <- lookupWithConstraint(x => {
+        val include: Rep[Boolean] = true
+
+        val userIdFilter = queryParameters.userId.map(userId => x.userId === userId)
+        val keyFilter = queryParameters.key.map(key => x.key === key)
+
+        val optionalFilters: List[Option[Rep[Boolean]]] = List(userIdFilter, keyFilter)
+        val filters = optionalFilters.map(_.getOrElse(include))
+        filters.reduceLeftOption(_ && _).getOrElse(include)
+      })
+      valueFilter = (userKeyValuePair: UserKeyValuePairWithId) => queryParameters.value map { _.equals(userKeyValuePair.userKeyValuePair.keyValuePair.value) } getOrElse true
```

Comment:
Because they come back from the DB encrypted. We have to wait for lookupWithConstraint to decrypt the values before we can filter.